### PR TITLE
Add support for french language, needs discussion.

### PR DESCRIPTION
**Pull request by MonsieurMan**
***Reference to #526***
This pull request adds support for the french language.  
I need your help on this issue though as I'm unsure about what some settings do : #1130

### CODE REVIEW

**Comment by jens1o** [comment]
I hope you now understand what we're delivering with these settings. Do not hesitate to ask if you have additional questions :)

File: src/i18n/langResources/lang.fr.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { ILangResource } from '../../models/i18n';
+import { constants } from '../../constants';
+
+export const langFr: ILangResource = {
+  newVersion: `Bienvenue dans la nouvelle version de ${constants.extensionName}.`,
+  seeReleaseNotes: 'Information à propos des derniers changement',
+  dontShowThis: 'Ne pas afficher ce message la prochaine fois',
+  seeReadme: 'A propos de l\'extension',
```
Comment (jens1o) [inline src/i18n/langResources/lang.fr.ts]: Can you please add something like `(anglais)` to make clear that this readme file is not in French?

File: src/i18n/langResources/lang.fr.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  welcome: `${constants.extensionName} supporte désormais l'API officielle.`,
+  activate: 'Activer',
+  aboutOfficialApi: 'Plus à propos des icônes de fichier & dossier.',
```
Comment (jens1o) [inline src/i18n/langResources/lang.fr.ts]: same here

File: package.nls.fr.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
   "configuration.customIconFolderPath.description": "Le chemin du dossier contenant les icônes personnalisées sur votre ordinateur.",
   "configuration.associations.files.description": "Associations personnalisées pour les icônes de fichier. (Ecrase les associations par défaut)",
   "configuration.associations.folders.description": "Associations personnalisées pour les icônes de dossier. (Ecrase les associations par défaut)",
-  "configuration.associations.fileDefault.file.description": "Définir l'icône de fichier sombre par défaut.", //Not sure about what these do and how
+  "configuration.associations.fileDefault.file.description": "Définir l'icône de fichier sombre par défaut. ",
```
Comment (jens1o) [inline package.nls.fr.json]: Please remove the whitespace at the end.

Comment (jens1o) [inline package.nls.fr.json]: I was too slow. 😆

Comment (MonsieurMan) [inline package.nls.fr.json]: Haha :joy:

**Comment by jens1o** [comment]
Looks good to me. Waiting for another review...

**Comment by jens1o** [comment]
If you want to, you can add `French` here. https://github.com/vscode-icons/vscode-icons/blob/4d382f12dca035ae3ebd8f1bb7468fa002a63be6/README.md#contributing-with-translations

**Comment by JimiC** [comment]
@MonsieurMan Awesome! Just hold the PR a little longer as we may need to do some modifications to it based on your suggestion.

**Comment by robertohuertasm** [comment]
Holding this until #1130 is closed.

**Pull request merged**